src/Compatibility/Microsoft.DotNet.PackageValidation/Checker.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    internal class Checker
```

Comment:
Is this really needed? The DiagnosticBag should already filter by noWarn and referenceId, diagnosticId tuple.

Reply:
I am working on assimilating it in the DiagnosticBag but we will need some changes to diagnostics bag.

Reply:
> I am working on assimilating it in the DiagnosticBag but we will need some changes to diagnostics bag.

That's fine. Let's discuss it on our meeting later today.